registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java

```diff
@@ -0,0 +1,391 @@
+        state.clearSeat();
+        state.addAllSeat(newSeats);
+        incrementState(state.build());
+    }
+
+    private boolean isOrderPartiallyReserved(final List<SeatQuantity> reservedSeats) {
+        final List<SeatQuantity> requestedSeats = getState().getSeatList();
+        if (reservedSeats.size() < requestedSeats.size()) {
+            return true;
+        }
+        final Collection<SeatQuantity> partlyReservedSeats = findPartlyReservedSeats(requestedSeats, reservedSeats);
+        final boolean isPartlyReserved = partlyReservedSeats.size() > 0;
+        return isPartlyReserved;
+    }
+
+    private static Collection<SeatQuantity> findPartlyReservedSeats(final List<SeatQuantity> requestedSeats,
+                                                                    final List<SeatQuantity> reservedSeats) {
+        final Collection<SeatQuantity> result = filter(requestedSeats, new Predicate<SeatQuantity>() {
+            @Override
+            public boolean apply(@Nullable SeatQuantity requestedOne) {
+                if (requestedOne == null) {
+                    return false;
+                }
+                final SeatQuantity reservedOne = findById(reservedSeats, requestedOne.getSeatTypeId());
+                if (reservedOne == null) {
+                    return false;
+                }
+                final boolean isPartlyReserved = reservedOne.getQuantity() < requestedOne.getQuantity();
+                return isPartlyReserved;
+            }
+        });
+        return result;
+    }
+
+    @Nullable
+    private static SeatQuantity findById(final List<SeatQuantity> seats, final SeatTypeId id) {
+        final SeatQuantity defaultValue = null;
+        final SeatQuantity result = find(seats, new Predicate<SeatQuantity>() {
+            @Override
+            public boolean apply(@Nullable SeatQuantity seat) {
+                final boolean result =
+                        (seat != null) &&
+                        seat.hasSeatTypeId() &&
+                        seat.getSeatTypeId().equals(id);
+                return result;
+            }
+        }, defaultValue);
+        return result;
+    }
+
+    @Override
+    @SuppressWarnings("RefusedBequest") // method from superclass does nothing
+    protected void validate(Order newState) throws IllegalStateException {
+        final int version = getVersion();
+        if (version > 0) {
+            Validator.checkNewState(newState);
+        }
+    }
+
+    private OrderTotalsCalculated buildOrderTotalsCalculated(OrderId orderId, ConferenceId conferenceId, List<SeatQuantity> seats) {
+        final OrderTotal total = pricingService.calculateTotalOrderPrice(conferenceId, seats);
+        return EventBuilder.buildOrderTotalsCalculated(orderId, total);
+    }
+
+    private static class EventBuilder {
+
+        private static OrderTotalsCalculated buildOrderTotalsCalculated(OrderId orderId, OrderTotal total) {
```

Comment:
Having 'build' or 'create' prefix makes it harder to read: all of the methods start from the same prefix.

Why not just orderTotalsCalculated(), orderPlaced(), etc.?